sketch_main.cpp

```diff
@@ -299,69 +244,12 @@ int main(int argc, char *argv[]) {
-                return min_hash.compute(seq);
-            };
-        } else if (FLAGS_sketch_method == "WMH") {
-            sketcher = [&,
-                        wmin_hash
-                        = WeightedMinHash<uint64_t>(kmer_word_size, FLAGS_embed_dim, FLAGS_max_len,
-                                                    HashAlgorithm::uniform, rd())](
-                               const std::vector<uint64_t> &seq) mutable {
-                return wmin_hash.compute(seq);
-            };
-        } else if (FLAGS_sketch_method == "OMH") {
-            sketcher = [&,
-                        omin_hash = OrderedMinHash<uint64_t>(kmer_word_size, FLAGS_embed_dim,
-                                                             FLAGS_max_len, FLAGS_tuple_length,
-                                                             HashAlgorithm::uniform, rd())](
-                               const std::vector<uint64_t> &seq) mutable {
-                return omin_hash.compute(seq);
-            };
-        }
-        std::function<Vec2D<double>(const std::vector<uint64_t> &)> slide_sketcher
-                = [&](const std::vector<uint64_t> & /*unused*/) { return new2D<double>(0, 0); };
-        SketchHelper<uint8_t, uint64_t, uint64_t> sketch_helper(sketcher, slide_sketcher);
-        sketch_helper.read_input();
-        sketch_helper.compute_sketches();
-        sketch_helper.save_output();
-    } else if (FLAGS_sketch_method.rfind("TS", 0) == 0) {
-        Tensor<uint64_t> tensor_sketch(kmer_word_size, FLAGS_embed_dim, FLAGS_tuple_length, rd());
-        TensorBlock<uint64_t> tensor_block(kmer_word_size, FLAGS_embed_dim, FLAGS_tuple_length,
-                                           FLAGS_block_size, rd());
-        TensorSlide<uint64_t> tensor_slide(kmer_word_size, FLAGS_embed_dim, FLAGS_tuple_length,
-                                           FLAGS_window_size, FLAGS_stride, rd());
-        std::function<std::vector<double>(const std::vector<uint64_t> &)> sketcher
-                = [&](const std::vector<uint64_t> &seq) {
-                      return FLAGS_block_size == 1 ? tensor_sketch.compute(seq)
-                                                   : tensor_block.compute(seq);
-                  };
-        std::function<Vec2D<double>(const std::vector<uint64_t> &)> slide_sketcher
-                = [&](const std::vector<uint64_t> &seq) { return tensor_slide.compute(seq); };
-        SketchHelper<uint8_t, uint64_t, double> sketch_helper(sketcher, slide_sketcher);
-        sketch_helper.read_input();
-        FLAGS_sketch_method == "TSS" ? sketch_helper.compute_slide()
-                                     : sketch_helper.compute_sketches();
-        sketch_helper.save_output();
-    } else {
-        std::cerr << "Unkknown method: " << FLAGS_sketch_method << std::endl;
-        exit(1);
-    }
+    std::cerr << "Unknown action: " << FLAGS_action << "\n";
```

Comment:
do you plan to add other actions? if not, maybe better to just keep things simple and remove the flag. In any case, it makes sense to default the flag value to "triangle" now.

Reply:
I have some more actions locally, but not yet sure how useful they are:
- normalize, which reads a triangle, divides by max lengths and prints it
- lengths, which just prints a list of all sequence lengths
- sketch, which currently writes a .sketch file per fasta input file. But I don't read them yet, and for now sketching is usually fast enough anyway